include/dylib.hpp

```diff
@@ -92,56 +101,21 @@ class dylib {
         return *this;
     }
 
-    dylib() noexcept = default;
-
     /**
      *  Creates a dynamic library instance
      *
-     *  @param path path to the dynamic library to load
-     *  @param ext use dylib::extension to specify the os extension (optional parameter)
+     *  @param path the path where is located the dynamic library you want to load
+     *  @param name the name of the dynamic library to load
+     *  @param decorations add os decorations to the library name
      */
-    explicit dylib(const char *path) {
-        open(path);
-    }
-
-    explicit dylib(const std::string &path) {
-        open(path.c_str());
-    }
-
-    dylib(const std::string &path, const char *ext) {
-        open(path, ext);
+    dylib(const std::string &path, const std::string &name, bool decorations = true) {
+        open(path, name, decorations);
     }
 
     ~dylib() {
         close();
```

Comment:
Drop the close function and inline its contents here. There's no sense in closing before destruction...

Reply:
Done